pkg/R/sparkR.R

```diff
@@ -36,14 +36,17 @@ sparkR.stop <- function(env) {
   if (exists(".sparkRjsc", envir = env)) {
     sc <- get(".sparkRjsc", envir = env)
     callJMethod(sc, "stop")
+    rm(".sparkRjsc", envir = env)
   }
 
-  callJStatic("SparkRHandler", "stopBackend")
-  # Also close the connection and remove it from our env
-  conn <- get(".sparkRCon", env)
-  close(conn)
-  rm(".sparkRCon", envir = env)
-
+  if(exists(".sparkRCon", envir = env)){
+    callJStatic("SparkRHandler", "stopBackend")
+    # Also close the connection and remove it from our env
+    conn <- get(".sparkRCon", env)
+    close(conn)
+    rm(".sparkRCon", envir = env)
+  }
+  
   # Finally, sleep for 1 sec to let backend finish exiting.
   # Without this we get port conflicts in RStudio when we try to 'Restart R'.
   Sys.sleep(1)
```

Comment:
What would the benefit be?  If the goal is to have everything shut down completely before going to the next step, putting the sys.sleep outside of the function makes a lot of sense.  If nothing else, I think it that makes the code much more intuitive, and does so at little to no cost. 

What do you think?